sbi/inference/snle/snle_base.py

```diff
@@ -88,25 +69,18 @@ def __init__(
             show_round_summary=show_round_summary,
         )
 
+        # As detailed in the docstring, density estimator is either a string or
+        # a callable. The function creating the neural network is attached to
+        # `_build_neural_net`. It will be called in the first round and receive
+        # thetas and xs as inputs, so that they can be used for shape inference and
+        # potentially for z-scoring.
         if isinstance(density_estimator, str):
-            density_estimator = utils.likelihood_nn(
-                model=density_estimator,
-                theta_shape=self._prior.sample().shape,
-                x_o_shape=self._x_shape,
-            )
-
-        # Create neural posterior which can sample().
-        # TODO Notice use of `snle_a`, OK so long as it is the sole descendant.
-        self._posterior = NeuralPosterior(
-            method_family="snle_a",
-            neural_net=density_estimator,
-            prior=self._prior,
-            x_shape=self._x_shape,
-            mcmc_method=mcmc_method,
-            get_potential_function=PotentialFunctionProvider(),
-        )
-
-        self._posterior.net.train(True)
+            self._build_neural_net = utils.likelihood_nn(model=density_estimator)
+        else:
+            self._build_neural_net = density_estimator
+        self._posterior = None
+        self._sample_with_mcmc = True
```

Comment:
Yes, for consistency e.g. with SNPE methods

Reply:
I mean if it should not be _made_ a class variable (not sure I got it a cross, must use verbs more often).

Reply:
Ah, well, technically it could be made a class variable but we are not using class variables anywhere so far and in SNPE it's set in the `__init__` constructor as well (needs to be). For consistency, I'd leave as is